cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/ControlDependenceGraphPass.kt

```diff
@@ -262,43 +274,50 @@ open class ControlDependenceGraphPass(ctx: TranslationContext) : EOGStarterPass(
  * Returns the updated state and true because we always expect an update of the state.
  */
 fun handleEdge(
-    currentEdge: Edge<Node>,
-    currentState: State<Node, IdentityHashMap<Node, IdentitySet<Node>>>,
-): State<Node, IdentityHashMap<Node, IdentitySet<Node>>> {
+    lattice: Lattice<PrevEOGStateElement>,
+    currentEdge: EvaluationOrder,
+    currentState: PrevEOGStateElement,
+): PrevEOGStateElement {
+    val lattice = lattice as? PrevEOGState ?: return currentState
+    var newState = currentState
+
+    val currentStart = currentEdge.start
+
     // Check if we start in a branching node and if this edge leads to the conditional
     // branch. In this case, the next node will move "one layer downwards" in the CDG.
-    if (currentEdge.start is BranchingNode) { // && currentEdge.isConditionalBranch()) {
+    if (
+        currentStart is BranchingNode ||
```

Comment:
Again, seeing this I am hoping that we can adequately integrate comprehension nodes by making them implement the `BranchingNode` interface but I don't know if that is sufficient.

Reply:
I added issue #2027